buttercup.el

```diff
@@ -1536,46 +1548,55 @@ EVENT and ARG are described in `buttercup-reporter'."
     (pcase event
       (`buttercup-started
        (setq buttercup-reporter-batch--start-time (float-time)
-             buttercup-reporter-batch--failures nil)
+             buttercup-reporter-batch--failures nil
+             ;; Don't put any output on hold if nothing is ever omitted.
+             buttercup--pending-output (not buttercup-silent-skipping))
        (let ((defined (buttercup-suites-total-specs-defined arg))
              (pending (buttercup-suites-total-specs-pending arg)))
          (if (> pending 0)
              (buttercup--print "Running %s out of %s specs.\n\n"
                                (- defined pending)
                                defined)
-           (buttercup--print "Running %s specs.\n\n" defined))))
+           (buttercup--print "Running %s specs.\n\n" defined)))
+       (buttercup--flush-pending-output))
 
       (`suite-started
        (let ((level (length (buttercup-suite-or-spec-parents arg))))
+         (buttercup--start-output-level)
          (buttercup--print "%s%s\n"
                            (make-string (* 2 level) ?\s)
                            (buttercup-suite-description arg))))
 
       (`spec-started
        (let ((level (length (buttercup-suite-or-spec-parents arg))))
+         (buttercup--start-output-level)
          (buttercup--print "%s%s"
                            (make-string (* 2 level) ?\s)
                            (buttercup-spec-description arg))))
 
       (`spec-done
-       (cond
-        ((eq (buttercup-spec-status arg) 'passed)) ; do nothing
-        ((eq (buttercup-spec-status arg) 'failed)
-         (buttercup--print "  FAILED")
-         (setq buttercup-reporter-batch--failures
-               (append buttercup-reporter-batch--failures
-                       (list arg))))
-        ((eq (buttercup-spec-status arg) 'pending)
-         (buttercup--print "  %s" (buttercup-spec-failure-description arg)))
-        (t
-         (error "Unknown spec status %s" (buttercup-spec-status arg))))
-       (buttercup--print " (%s)\n"
-                         (seconds-to-string
-                          (float-time (buttercup-elapsed-time arg)))))
+       (if (buttercup-spec-omitted-p arg)
+           (buttercup--discard-one-output-level)
```

Comment:
It should be possible to move this into the cond as the first case. Then you wouldn't need any reindentation of the rest of the code.

Reply:
No, it's not possible because the structure is `(if X (Y) (cond ...) (Z))`. I.e. all `cond` clauses lead to `(Z)` branch, but `(Y)` must not.